types/react-native-fbsdk/index.d.ts

```diff
@@ -0,0 +1,1089 @@
+         */
+        setShouldFailOnDataError(shouldFailOnDataError: boolean): void;
+    }
+
+    const MessageDialog: MessageDialog;
+
+    interface SendButtonProps {
+        /**
+         * Content to be shared.
+         */
+        shareContent: ShareContent;
+
+        /**
+         * View style, if any.
+         */
+        style?: StyleSheet.Style;
+    }
+
+    class SendButton extends Component<SendButtonProps, any> {}
+
+    interface ShareApi {
+        /**
+         * Check if the content can be shared via share api.
+         */
+        canShare(shareContent: ShareContent): Promise<boolean>;
+
+        /**
+         * For iOS only, creates a User Owned Open Graph object without an action.
+         * NOTE: Only one share action can be performed at a time.
+         * @platform ios
+         */
+        createOpenGraphObject(openGraphObject: ShareOpenGraphObject): Promise<any>;
+
+        /**
+         * Shares the specified content with a message.
+         * NOTE: Only one share action can be performed at a time.
+         */
+        share(shareContent: ShareContent, graphNode: string, message: string): Promise<any>;
+    }
+
+    const ShareApi: ShareApi;
+
+    interface ShareButtonProps {
+        /**
+         * Content to be shared.
+         */
+        shareContent: ShareContent;
+
+        /**
+         * View style, if any.
+         */
+        style?: StyleSheet.Style;
+    }
+
+    class ShareButton extends Component<ShareButtonProps, any> {}
+
+    type ShareDialogMode = ShareDialogModeIOS | ShareDialogModeAndroid;
+    type ShareDialogModeAndroid =
+        /**
+         * The mode is determined automatically.
+         */
+        'automatic'|
+        /**
+         * The native dialog is used.
+         */
+        'native'|
+        /**
+         * The web dialog is used.
+         */
+        'web'|
+        /**
+         * The feed dialog is used.
+         */
+        'feed';
+    type ShareDialogModeIOS =
+        /*
+        * Acts with the most appropriate mode that is available.
+        */
+        'automatic'|
+        /*
+        * Displays the dialog in Safari.
+        */
+        'browser'|
+        /*
+        * Displays the dialog in a UIWebView within the app.
+        */
+        'webview';
+
+    interface ShareDialog {
+        /**
+         * Check if the dialog can be shown.
+         */
+        canShow(shareContent: ShareContent): Promise<boolean>;
+
+        /**
+         * Shows the dialog using the specified content.
+         */
+        show(shareContent: ShareContent): Promise<any>;
+
+        /**
+         * Sets the mode for the share dialog.
+         */
+        setMode(mode: ShareDialogMode): void;
+
+        /**
+         * Sets whether or not the native share dialog should fail when it encounters a data error.
+         */
+        setShouldFailOnDataError(shouldFailOnDataError: boolean): void;
+    }
+
+    const ShareDialog: ShareDialog;
```

Comment:
Especially if it can't be instantiated. However this is fine if the interface needs to be implemented or referenced somewhere as a type. Is this interface being used to organize (in which case, make it a namespace), or not?